optuna/samplers/_tpe/sampler.py

```diff
@@ -905,6 +939,9 @@ def _get_observation_pairs(
                     score = (-step, sign * intermediate_value)
             else:
                 score = (float("inf"), 0.0)
+        elif trial.state is TrialState.RUNNING:
+            assert constant_liar
+            score = (-float("inf"), sign * float("inf"))
```

Comment:
The size of the constant is fixed to infinity, but it could be made adjustable (not a strong opinion).

Reply:
I agree. I tried to address that in the description.